flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/converters/SqlCreateModelConverter.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.validate.SqlValidator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+
+/** A converter for {@link org.apache.flink.sql.parser.ddl.SqlCreateModel}. */
+public class SqlCreateModelConverter implements SqlNodeConverter<SqlCreateModel> {
+
+    @Override
+    public Operation convertSqlNode(SqlCreateModel sqlCreateModel, ConvertContext context) {
+        UnresolvedIdentifier unresolvedIdentifier =
+                UnresolvedIdentifier.of(sqlCreateModel.fullModelName());
+        ObjectIdentifier identifier =
+                context.getCatalogManager().qualifyIdentifier(unresolvedIdentifier);
+        Map<String, String> modelOptions = getModelOptions(sqlCreateModel);
+
+        ModelSchemaBuilderUtils schemaBuilderUtil =
+                new ModelSchemaBuilderUtils(
+                        context.getSqlValidator(),
+                        SqlNode::toString,
+                        context.getCatalogManager().getDataTypeFactory());
+
+        CatalogModel catalogModel =
+                CatalogModel.of(
+                        schemaBuilderUtil.getSchema(sqlCreateModel.getInputColumnList()),
+                        schemaBuilderUtil.getSchema(sqlCreateModel.getOutputColumnList()),
+                        modelOptions,
+                        sqlCreateModel.getComment().map(SqlLiteral::toValue).orElse(null));
+
+        return new CreateModelOperation(
+                identifier,
+                context.getCatalogManager().resolveCatalogModel(catalogModel),
+                sqlCreateModel.isIfNotExists(),
+                sqlCreateModel.isTemporary());
+    }
+
+    private Map<String, String> getModelOptions(SqlCreateModel sqlCreateModel) {
+        Map<String, String> options = new HashMap<>();
+        sqlCreateModel
+                .getPropertyList()
+                .getList()
+                .forEach(
+                        p ->
+                                options.put(
+                                        ((SqlTableOption) Objects.requireNonNull(p)).getKeyString(),
+                                        ((SqlTableOption) p).getValueString()));
+        return options;
+    }
+
+    /** Builder for {@link Schema} of a model. */
+    private static class ModelSchemaBuilderUtils extends SchemaBuilderUtil {
+        ModelSchemaBuilderUtils(
+                SqlValidator sqlValidator,
+                Function<SqlNode, String> escapeExpressions,
+                DataTypeFactory dataTypeFactory) {
+            super(sqlValidator, escapeExpressions, dataTypeFactory);
+        }
+
+        private Schema getSchema(SqlNodeList nodeList) {
+            columns.clear();
+            for (SqlNode column : nodeList) {
+                if (column instanceof SqlRegularColumn) {
+                    SqlRegularColumn regularColumn = (SqlRegularColumn) column;
+                    columns.put(
```

Comment:
I think we should check whether the column is duplicate. If input schema contains columns with same name, please throw exception to nodify users.